src/components/SpacemeshApiClientMock/index.tsx

```diff
@@ -0,0 +1,231 @@
+import * as React from 'react';
+import { ISpacemeshApi } from '../../model';
+import { ITransaction, TransactionStatus, MockData} from 'src/types';
+
+interface IProps extends ISpacemeshApi{
+    apiReady: boolean;
+}
+
+interface IState {
+    connectedToApi: boolean;
+}
+
+const SpacemeshApiClientMock = <P extends IProps>(Component: React.ComponentType<P>) => {
+    return class SpacemeshApiClient extends React.Component<P & IProps, IState> implements ISpacemeshApi {
```

Comment:
Clean code and good documentation, thanks.

I came back here for a second pass and am missing the class-level comment :-(

Reply:
@daonb What do you mean by the class-level comment?